libs/sdk-common/src/input_parser.rs

```diff
@@ -175,6 +175,27 @@ pub async fn parse(input: &str) -> Result<InputType> {
         };
     }
 
+    // Covers BIP 21 URIs and simple onchain Liquid addresses (which are valid BIP 21 with the 'liquid:' or 'liquidtestnet:' prefix)
+    if let Ok(liquid_address_data) = parse_liquid_bip21(input) {
+        // Special case of LN BOLT11 with onchain fallback
+        // Search for the `lightning=bolt11` param in the BIP21 URI and, if found, extract the bolt11
+        let mut invoice_param: Option<LNInvoice> = None;
+        if let Some(query) = input.split('?').collect::<Vec<_>>().get(1) {
```

Comment:
NIT:
I believe we can depend on string slicing here instead of collecting to a vector to avoid unnecessary allocations.
```rust
        if let Some(query) = input.find('?') {
            let invoice_params = &input[query + 1..];
            ...
        }
```

We can also return early instead of matching `invoice_param` later on.

Performance gains are may appear negligible but `parse` is one of the highest usage rate API's we have, both internally and externally, so even the smallest optimization adds up.